pkg/controller/operators/catalog/operator.go

```diff
@@ -2312,7 +2319,26 @@ func (o *Operator) ExecutePlan(plan *v1alpha1.InstallPlan) error {
 					// Ensure Unstructured Object
 					status, err := ensurer.EnsureUnstructuredObject(resourceInterface, unstructuredObject)
 					if err != nil {
-						return err
+						if step.Optional {
+							switch k8serrors.ReasonForError(err) {
+							case
+								metav1.StatusReasonUnauthorized,
```

Comment:
Do we handle the case of `StatusReasonAlreadyExists` somewhere? Or do we need to?

Reply:
True, this deviates from the EP. There should not be an existing resource (I don't see a reason for it) with same namespace/name when the operator gets installed. Failing the InstallPlan would give the opportunity to hte cluster admin to remove it and retry the installation. This would guarantee that the available resource is the intended one. In that case I would update the EP and align it with the implementation. Let me know if you think that it should be done differently.